### DISCUSSION TITLE
Clustering Data Kategorik

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#691] [Q&A] [answered]
Selamat siang kak. Saya ingin melakukan clustering pada data yang keseluruhannya berupa kategorik saja. Apakah ada referensi metode yang dapat saya gunakan ? 

Mohon bantuannya kak, terima kasih 🙏

**Comment by kinantytasyao** [comment, marked as answer]
Dear @yohanespm77 

Jika kita memiliki semua variabel dalam tipe kategorik maka kita dapat menggunakan metode K-modes untuk clustering datanya. Berikut referensi yang bisa diperdalam untuk mempelajari K-modes:

- K-modes in R: https://www.rdocumentation.org/packages/klaR/versions/1.7-2/topics/kmodes
- K-modes in Python: https://www.analyticsvidhya.com/blog/2021/06/kmodes-clustering-algorithm-for-categorical-data/

Demikian yang dapat kami sampaikan, semoga membantu. Terima kasih.

**Reply by yohanespm77** [reply]
Kalo untuk data yang berupa character, berarti harus dirubah menjadi factor dulu ya? Misalnya account_id

**Reply by kinantytasyao** [reply]
Benar Pak Yohanes, dan biasanya account_id ini adalah id yang unik sehingga nilainya tidak berulang dan tidak disarankan untuk dijadikan variabel kategorikal

**Reply by yohanespm77** [reply]
Berarti tidak perlu diubah menjadi kategorik kak ? Apakah modelnya bisa menjalankan tipe data character jg kak ?

**Reply by kinantytasyao** [reply]
Bukan Pak, tetapi tidak disarankan memasukkan variabel account_id ke clusteringnya. Namun jika memang masih ingin memasukkan account_id ke dalam model maka perlu dilakukan pengubahan tipe data menjadi factor terlebih dahulu
